### PR TITLE
net: conntrack improvements

### PR DESCRIPTION
**Pull request by mmatczuk**

### CODE REVIEW

File: utils/reflectx/reflectx_test.go [hunk 1]
```diff
@@ -1,26 +1,23 @@
 // Copyright 2022-2024 Sauce Labs Inc., all rights reserved.
 //
-// Copyright 2015 Google Inc. All rights reserved.
```
Comment (mmatczuk) [inline utils/reflectx/reflectx_test.go:3]: This is not Google's code.

File: testdata/TestDialerMetrics_traffic.golden.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# HELP test_dialer_closed_total Number of closed connections
+# TYPE test_dialer_closed_total counter
+test_dialer_closed_total{host="localhost"} 10
+# HELP test_dialer_dialed_total Number of dialed connections
+# TYPE test_dialer_dialed_total counter
+test_dialer_dialed_total{host="localhost"} 10
```
Comment (Choraden) [inline testdata/TestDialerMetrics_traffic.golden.txt:6]: nit: Golden files of TestDialerMetrics `traffic` and `default` are identical. This does not really tests if returned connection tracks traffic.

File: utils/reflectx/reflectx_test.go [hunk 3]
```diff
@@ -1,26 +1,23 @@
...
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-package martian
+package reflectx_test
```
Comment (Choraden) [inline utils/reflectx/reflectx_test.go:7]: Is a separate test package intentional? I don't recall us using this practice.

**Comment by Choraden** [comment]
LGTM, left minor comments.

File: utils/reflectx/reflectx_test.go (hunk 3 above)
Comment (mmatczuk) [reply utils/reflectx/reflectx_test.go:7]: Moved tests to `reflectx` package

**Comment by mmatczuk** [comment]
@Choraden PTAL

**Review by Choraden — APPROVED** [review]

**Pull request merged**